components/hdbpp-health-check/src/HealthCheck.cpp

```diff
@@ -69,29 +69,33 @@ std::tuple<HealthCheck::HealthCheckResult, std::string> HealthCheck::check_hosts
         rapidjson::Document document;
         document.Parse(result->body.c_str());
 
-        std::tuple<HealthCheck::HealthCheckResult, std::string> result;
-
-        if (!document.IsObject() || !document["state"].IsString() || document.HasMember("state"))
-            result = std::make_tuple(HealthCheckResult::ConnectionProblem, invalid_response);
+        if (!document.IsObject() || !document.HasMember("state") || !document["state"].IsString())
+            return std::make_tuple(HealthCheckResult::ConnectionProblem, invalid_response);
+
+        // Retrieve the error message if there is any
+	std::string error_message;
+	if (document.HasMember("message") && document["message"].IsString())
+            error_message = "\n" + std::string(document["message"].GetString());
+	
+        else
```

Comment:
Couple of Clang failures:

/users/sjames/workspace/projects/HdbppTimescale/projects/hdbpp-timescale-project/components/hdbpp-health-check/src/HealthCheck.cpp:80:9: warning: different indentation for 'if' and corresponding 'else' [readability-misleading-indentation]
        else